selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/AbstractTestWorkspaceServiceClient.java

```diff
@@ -120,6 +121,27 @@ public void delete(String workspaceName, String userName) throws Exception {
 
     requestFactory.fromUrl(getIdBasedUrl(workspace.getId())).useDeleteMethod().request();
 
+    WaitUtils.waitSuccessCondition(
+        () -> {
+          try {
+            LOG.info(
+                "Check workspace name='{}', id='{}', username='{}' existence",
+                workspaceName,
+                workspace.getId(),
+                userName);
+            return !exists(workspaceName, userName);
+          } catch (Exception e) {
```

Comment:
Is it possible to set some particular exception?

Reply:
it is thrown by `exists()` itself